applications/graphQL/query.py

```
@@ -0,0 +1,96 @@
+            queryset = queryset.filter(category=category)
+        return queryset.order_by("?").first()
+
+    def resolve_insult_by_id(root, info, id: str) -> Insult:
+        """Resolve an insult by its ID."""
+        try:
+            return Insult.objects.get(id=id)
+        except ObjectDoesNotExist:
+            raise GraphQLError(f"Insult with ID {id} not found")
+
+    def resolve_insults_by_category(
+        root, info, category: str, offset: int = 0, limit: int = 10
```

Ruff (0.12.2)

Missing type annotation for function argument `info`

(ANN001)